### PR TITLE
LPS-182317 Implement utility page for internal server error

### PR DESCRIPTION
**Pull request by jkappler**
## Motivation

[Link to ticket](https://issues.liferay.com/browse/LPS-182317)

Adds utility page for error 500

## Rant

* It only works for guest site at the moment, since the error could be thrown anywhere and we don't have information where it happens
* Error has to be thrown by hand to make it work
* Everything is behind a FF, and component is disabled

## How to make it work 

* Enable ff for LPS-165914
* Enable component: `scr:enable com.liferay.layout.utility.page.status.internal.InternalServerErrorLayoutUtilityPageEntryViewRenderer`
* After this, the utility page templates for error 500 can be created through the interface

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by jkappler** [comment]
ci:test:sf

File: modules/apps/layout/layout-utility-page/layout-utility-page-status/src/main/resources/META-INF/resources/internal_server_error.jsp [hunk 1]
```diff
@@ -19,4 +19,18 @@
...
+		<liferay-ui:message key="internal-server-error" />
+	</h3>
+
+	<liferay-ui:message key="an-error-occurred-while-accessing-the-requested-resource" />
+
+	<br /><br />
+
+	<code class="lfr-url-error"><%= statusDisplayContext.getEscapedURL(themeDisplay) %></code>
+
+	  <%
```
Comment (ealonso) [inline modules/apps/layout/layout-utility-page/layout-utility-page-status/src/main/resources/META-INF/resources/internal_server_error.jsp:32]: @jkappler There is a SF issue right here

File: modules/apps/layout/layout-utility-page-api/src/main/java/com/liferay/layout/utility/page/converter/LayoutUtilityPageEntryTypeConverter.java [hunk 2]
```diff
@@ -49,6 +49,8 @@ public static String convertToInternalValue(String label) {
 			"Error", LayoutUtilityPageEntryConstants.TYPE_STATUS
 		).put(
 			"ErrorCode404", LayoutUtilityPageEntryConstants.TYPE_SC_NOT_FOUND
+		).put(
```
Comment (ealonso) [inline modules/apps/layout/layout-utility-page-api/src/main/java/com/liferay/layout/utility/page/converter/LayoutUtilityPageEntryTypeConverter.java:52]: @jkappler Another SF issue, fixed in commit 00708bcfb1e2893de1004de1b6b42181165216e1

File: modules/apps/layout/layout-utility-page/layout-utility-page-status/src/main/resources/META-INF/resources/internal_server_error.jsp (hunk 1 above)
Comment (jkappler) [reply modules/apps/layout/layout-utility-page/layout-utility-page-status/src/main/resources/META-INF/resources/internal_server_error.jsp]: Thanks @ealonso, didn't noticed

File: modules/apps/layout/layout-utility-page-api/src/main/java/com/liferay/layout/utility/page/converter/LayoutUtilityPageEntryTypeConverter.java (hunk 2 above)
Comment (jkappler) [reply modules/apps/layout/layout-utility-page-api/src/main/java/com/liferay/layout/utility/page/converter/LayoutUtilityPageEntryTypeConverter.java:52]: Thanks @ealonso, fixed

**Comment by jkappler** [comment]
ci:test:sf

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by alessandromalves** [comment]
Hey @jkappler, could you please take a look at those integration tests failures?

[com.liferay.adaptive.media.document.library.web.internal.counter.test.DLAMImageCounterTest](https://testray.liferay.com/home/-/testray/case_results/2298748336)

```[exec] com.liferay.adaptive.media.document.library.web.internal.counter.test.DLAMImageCounterTest > testDLAMImageCounterOnlyCountsDefaultRepositoryImages FAILED
     [exec]     java.lang.AssertionError: expected:<5> but was:<6>
     [exec]     	at org.junit.Assert.fail(Assert.java:89)
     [exec]     	at org.junit.Assert.failNotEquals(Assert.java:835)
     [exec]     	at org.junit.Assert.assertEquals(Assert.java:647)
     [exec]     	at org.junit.Assert.assertEquals(Assert.java:633)
     [exec]     	at com.liferay.adaptive.media.document.library.web.internal.counter.test.DLAMImageCounterTest.testDLAMImageCounterOnlyCountsDefaultRepositoryImages(DLAMImageCounterTest.java:85)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
@alessandromalves, @ealonso, sending without site initializer until Lima merges the test fix

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:forward

**Comment by jkappler** [comment]
tests don't seem related

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
@alessandromalves, I think we can fw it right?

**Comment by alessandromalves** [comment]
The failures are not related to this PR. See https://github.com/brianchandotcom/liferay-portal/pull/134731

**Pull request closed without merging**